compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1933,28 +1933,90 @@ TR::Register *OMR::X86::TreeEvaluator::integerDualMulEvaluator(TR::Node *node, T
 
 TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   TR::Register *targetRegister = 0;
-   TR::Node     *firstChild     = node->getFirstChild();
-   TR::Node     *secondChild    = node->getSecondChild();
-   bool          nodeIs64Bit    = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
-   intptrj_t    value;
+   TR::Register *targetRegister  = 0;
+   TR::Node     *firstChild      = node->getFirstChild();
+   TR::Node     *secondChild     = node->getSecondChild();
+   bool          nodeIs64Bit     = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
+   bool          isLoadConst     = secondChild->getOpCode().isLoadConst();
+   bool          isByteOperation = false;
+   bool          isMemRef        = false;
+   TR_X86OpCodes opCode          = BADIA32Op; 
+   TR_X86OpCodes regOpCode       = BADIA32Op; 
+   TR_X86OpCodes memOpCode       = BADIA32Op; 
+   TR_X86OpCodes movOpCode       = BADIA32Op;
+   int64_t       value;
 
    if (node->isDualCyclic())
       {
       return TR::TreeEvaluator::integerDualMulEvaluator(node, cg);
       }
 
-   if (secondChild->getOpCode().isLoadConst() &&
-      (value = TR::TreeEvaluator::integerConstNodeValue(secondChild, cg)))
+   // Opcode evaluation for bmul, smul imul and lmul
+   switch (secondChild->getDataType())
+      {
+      case TR::Int8:
+         isMemRef = secondChild->getOpCode().isMemoryReference(); 
```

Comment:
`value` is not assigned for `case TR::Int8:` and can have uninitialized value when `isLoadConst ` is true for `case TR::Int8:`. I think it's better to keep assigning value at the same place as before but replace the second condition into a query taking `TR::Int8` and `TR::int16` into consideration https://github.com/eclipse/omr/blob/ce7fdb230728b6460dad976959340c4c09a053de/compiler/x/codegen/BinaryEvaluator.cpp#L1947-L1948